### DISCUSSION TITLE
Huge thanks to emscripten!

### DISCUSSION THREAD

**Discussion by goldwaving** [emscripten-core/emscripten#16679] [Show and tell]
Just want to say how grateful I am for Emscripten and all the amazing work done by the contributors!

A few years ago, I started porting my Windows program to Android and iOS.  I quickly found these environments to be restrictive and needlessly complicated for app development (Android API version issues, iOS certificate/provisioning issues, store restrictions).  Shortly after I completed the Android version, the development tools provider I was using dropped support for Android leaving me no way to update the app.  I started looking into alternatives and eventually settled on Emscripten.  With Emscripten, VSCode, and a custom made build tool, I had a new development environment that was free and had the latest C++ support too.

It has taken 15 months to convert the program to a web app, but now I have something that runs on everything.  Finally we have a write once run everywhere development environment that never could have been possible without Emscripten.  Thanks again!

For anyone interested, you can find the app [here](https://goldwave.com/editor).

**Comment by stevedekorte** [comment]
I'd like to 2nd this comment. I got my C library working in a web page in a few days, and even the coroutines seem to work perfectly with a few code changes for emscripten fibers. I'm excited about what's possible with this platform and would like to make emscripten the default target for development and testing of my C projects... if I can find a way to get the VSCode debugger to step through C statements and view C stack frames instead of WASM instructions and stack.

**Reply by goldwaving** [reply]
Hopefully that is just a matter of time.  There is an extension for Chrome that supports C source debugging: https://developer.chrome.com/blog/wasm-debugging-2020/
